python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -324,12 +326,45 @@ def __init__(
         self.tp_size = (
             tp_size if tp_size is not None else get_tensor_model_parallel_world_size()
         )
+        self.tp_rank = get_tensor_model_parallel_rank()
+        self.num_experts = num_experts
+        self.reduce_results = reduce_results
+        self.expert_map = None
+        self.enable_flashinfer_moe = enable_flashinfer_moe
+        if self.enable_flashinfer_moe:
+            self.reduce_results = True
```

Comment:
@trevor-m Why we have to do reduce here? There is an reduce op at [here](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/deepseek_v2.py#L357). And this reduce op is surplus and shoud be removed.